hedera-node/src/main/java/com/hedera/services/txns/contract/process/CreateEvmTxProcessor.java

```diff
@@ -32,35 +33,34 @@
 import org.hyperledger.besu.evm.Code;
 import org.hyperledger.besu.evm.frame.MessageFrame;
 
+import javax.inject.Inject;
+import javax.inject.Singleton;
 import java.time.Instant;
 import java.util.Optional;
 
+@Singleton
 public class CreateEvmTxProcessor extends EvmTxProcessor {
 
-	private final Address newContractAddress;
-
+	@Inject
 	public CreateEvmTxProcessor(
+			HederaWorldState worldState,
 			HbarCentExchange exchange,
-			HederaWorldState.Updater worldUpdater,
 			UsagePricesProvider usagePrices,
-			GlobalDynamicProperties globalDynamicProperties,
-			Address newContractAddress
+			GlobalDynamicProperties globalDynamicProperties
 	) {
-		super(exchange, usagePrices, worldUpdater, globalDynamicProperties);
-		this.newContractAddress = newContractAddress;
+		super(worldState, exchange, usagePrices, globalDynamicProperties);
 	}
 
 	public TransactionProcessingResult execute(
 			final Account sender,
+			final Address receiver,
 			final long providedGasLimit,
 			final long value,
 			final Bytes code,
 			final Instant consensusTime) {
-
 		final long gasPrice = gasPriceTinyBarsGiven(consensusTime);
-		final long gasLimit = providedGasLimit > dynamicProperties.maxGas() ? dynamicProperties.maxGas() :
-				providedGasLimit;
-		return super.execute(sender, Optional.empty(), gasPrice, gasLimit, value, code, true, consensusTime);
+
+		return super.execute(sender, Optional.of(receiver), gasPrice, providedGasLimit, value, code, true, consensusTime, false);
```

Comment:
If we are always passing the `receiver` (in every case), wouldn't it be better to remove the `Optional` type?